src/frontend/src/pages/myfeeds.js

```diff
@@ -15,94 +34,312 @@ const useStyles = makeStyles((theme) => ({
+            }}
+          >
+            <Box p={2}>
+              <Typography align="center">
+                Please enter the web address of your blog&apos;s RSS or Atom feed.
+              </Typography>
+              <Typography align="center">
+                <em>
+                  (<a href="https://rss.com/blog/find-rss-feed/">Not sure how to find that?</a>)
+                </em>
+              </Typography>
+            </Box>
+          </Popover>
+        </div>
+      )}
+    </PopupState>
+  );
+}
+
+function DeleteDialogButton({ feed }) {
+  const classes = useStyles();
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const removeFeed = () => {
+    console.log(`Removing feed hosted at URL ${feed.url}`);
+    // TODO
+    window.location.reload(false);
+  };
+
+  return (
+    <div>
+      <IconButton classes={{ root: classes.button }} onClick={handleClickOpen}>
+        <Delete color="secondary" />
+      </IconButton>
+      <Dialog
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="alert-dialog-title"
+        aria-describedby="alert-dialog-description"
+      >
+        <DialogTitle id="alert-dialog-title">{`Remove feed hosted at ${feed.url}?`}</DialogTitle>
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">
+            Telescope will no longer display blog posts from this feed.
+          </DialogContentText>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleClose} color="secondary" variant="outlined" autoFocus>
+            Cancel
+          </Button>
+          <Button
+            onClick={() => {
+              handleClose();
+              removeFeed();
+            }}
+            color="primary"
+            variant="contained"
+          >
+            Confirm
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </div>
+  );
+}
+
+DeleteDialogButton.propTypes = {
+  feed: PropTypes.object,
+};
+
 export default function MyFeeds() {
   const classes = useStyles();
-  const [feedAuthor, setFeedAuthor] = useState('');
-  const [feedUrl, setFeedUrl] = useState('');
+  const [newFeedAuthor, setNewFeedAuthor] = useState('');
+  const [newFeedUrl, setNewFeedUrl] = useState('');
+  const [submitStatus, setSubmitStatus] = useState({ message: '', isError: false });
+  const [userInfo, setUserInfo] = useState({});
+  const [feedHash, updateFeedHash] = useState({}); // { id1: {author: '...', url: '...'} }, ... ]
+
   const { telescopeUrl } = useSiteMetadata();
 
-  function handleAuthorChange(author) {
-    setFeedAuthor(author);
-  }
+  const newFeedAuthorRef = React.createRef();
+  const newFeedUrlRef = React.createRef();
+
+  async function hashUserFeeds() {
+    const [user, feedItems] = await Promise.all([
+      fetch(`${telescopeUrl}/user/info`).then((res) => res.json()),
```

Comment:
All of this can/will fail at some point, we need proper error handling.  You can chat with @c3ho or @raygervais, who have recently done code like this with error handling.

Reply:
I also think we should probably move the call to get user info to the top level of this page, and already know who you are when you access the content.  That's something for #760.

Reply:
> All of this can/will fail at some point, we need proper error handling. You can chat with @c3ho or @raygervais, who have recently done code like this with error handling.

Although we've recognized the fetching done here is mostly a stop-gap for proper backend logic, I refactored the fetch requests a bit in https://github.com/Seneca-CDOT/telescope/pull/933/commits/2ae7efe11944d53d23e43c897c561cf9948cfedb, wrapping them in try/catch blocks similar to those I've observed @c3ho and @raygervais using in the backend.

> I also think we should probably move the call to get user info to the top level of this page, and already know who you are when you access the content. That's something for #760.

Agreed. For what it's worth, as another temporary measure, I separated the logic for fetching user-info from that of fetching feeds by placing it within a separate useEffects call (https://github.com/Seneca-CDOT/telescope/pull/933/commits/2ae7efe11944d53d23e43c897c561cf9948cfedb#diff-341423ac4fcd5b3e25e7dd93d2bae8e1R35)